### PR TITLE
passt:move utilities to provider

### PR DESCRIPTION
**Pull request by chloerh**
Move utilities to provider for future tests

Test result:
 (1/2) type_specific.io-github-autotest-libvirt.virtual_network.passt.attach_detach.ip_portfw.attach_detach.root_user: PASS (259.49 s)
 (2/2) type_specific.io-github-autotest-libvirt.virtual_network.passt.interface_function.ip_portfw.non_root_user: PASS (419.86 s)

### CODE REVIEW

File: provider/virtual_network/passt.py [hunk 1]
```diff
@@ -334,3 +368,34 @@ def check_port_listen(ports, protocol, host_ip=None):
         else:
             raise exceptions.TestFail(f'Address:Port {item} is NOT being '
                                       'listened to')
+
+
+def check_portforward(vm, host_ip, params):
```
Comment (dzhengfy) [inline provider/virtual_network/passt.py:373]: docstring, thanks.

File: provider/virtual_network/passt.py [hunk 2]
```diff
@@ -334,3 +368,34 @@ def check_port_listen(ports, protocol, host_ip=None):
...
+            conn_check_list.append(eval(v))
+    LOG.debug(f'conn_check_list: {conn_check_list}')
+
+    check_portforward_connetion(vm, conn_check_list,
+                                test_user=params.get('test_user'))
+
+
+def check_passt_pid_not_exist():
+    """
+    Check whether pid of passt disappeared, fail test if not
```
Comment (dzhengfy) [inline provider/virtual_network/passt.py:395]: We could have `:raise: exceptions.TestFail `

File: provider/virtual_network/passt.py [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
+    selinux_status = utils_selinux.get_status()
+    if selinux_status != 'enforcing':
+        utils_selinux.set_status('enforcing')
+    return selinux_status
+
+
+def check_socat_installed():
+    """
+    Check whether socat is installed, fail test if not
+    """
```
Comment (dzhengfy) [inline provider/virtual_network/passt.py:33]: We could have `:raise: exceptions.TestError`

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Review by dzhengfy — APPROVED** [review]

File: provider/virtual_network/passt.py (hunk 1 above)
Comment (chloerh) [reply provider/virtual_network/passt.py:375]: Done

File: provider/virtual_network/passt.py (hunk 2 above)
Comment (chloerh) [reply provider/virtual_network/passt.py]: Done

File: provider/virtual_network/passt.py (hunk 3 above)
Comment (chloerh) [reply provider/virtual_network/passt.py:35]: Done

**Review by nanli1 — APPROVED** [review]

**Pull request merged**